pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -0,0 +1,343 @@
+		//create a configMap with no values
+		current = corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      GetSplunkMonitoringconsoleConfigMapName(namespace, SplunkMonitoringConsole),
+				Namespace: namespace,
+			},
+			Data: make(map[string]string),
+		}
+
+		err = splctrl.CreateResource(client, &current)
+		if err != nil {
+			return nil, err
+		}
+		return &current, nil
+	}
+
+	if err == nil {
+		revised := current.DeepCopy()
+		for _, url := range newURLs {
+			_, ok := revised.Data[url.Name]
+			if addNewURLs { //if new instances are coming up
+				if !ok { //key doesn't exist in the configMap, then add it
+					revised.Data[url.Name] = url.Value //if new url doesn't exist add it in configmap
+				} else {
+					//TODO: For "SEARCH_HEAD_URL" need to find a better approach to update configMap, any other data structure?
+					if url.Name == "SPLUNK_SEARCH_HEAD_URL" {
+						if revised.Data[url.Name] == url.Value {
+							break
+						}
+						//delete all URLs for the crName custom resource
+						currentURLs := strings.Split(revised.Data[url.Name], ",") //split the existing values
+						for _, curr := range currentURLs {
+							if strings.Contains(curr, crName) { //delete that entry
+								revised.Data[url.Name] = strings.ReplaceAll(revised.Data[url.Name], curr, "")
+								if strings.HasPrefix(revised.Data[url.Name], ",") {
+									res := revised.Data[url.Name]
+									revised.Data[url.Name] = strings.TrimPrefix(res, ",")
+								}
+								if strings.HasSuffix(revised.Data[url.Name], ",") {
+									res := revised.Data[url.Name]
+									revised.Data[url.Name] = strings.TrimSuffix(res, ",")
+								}
+								if strings.Contains(revised.Data[url.Name], ",,") {
+									res := revised.Data[url.Name]
+									revised.Data[url.Name] = strings.ReplaceAll(res, ",,", ",")
+								}
+							}
+						}
+						newInsURLs := strings.Split(url.Value, ",")
+						for _, newEntry := range newInsURLs {
+							if !(strings.Contains(revised.Data[url.Name], newEntry)) {
+								if revised.Data[url.Name] == "" {
+									revised.Data[url.Name] = newEntry
+								} else {
+									str := []string{revised.Data[url.Name], newEntry}
+									revised.Data[url.Name] = strings.Join(str, ",")
+								}
+							}
+						}
+					} else {
+						//now add the incoming URLs as fresh entry
+						newInsURL := strings.Split(url.Value, ",")
+						for _, newEntry := range newInsURL {
+							if !(strings.Contains(revised.Data[url.Name], newEntry)) {
+								str := []string{revised.Data[url.Name], newEntry}
+								revised.Data[url.Name] = strings.Join(str, ",")
+							}
+						}
+					}
+				}
+			} else { //removing custom resource. Update configMap
+				//check if only dummy entery exist then no deletion can happen
+				if strings.Contains(revised.Data[url.Name], url.Value) {
+					revised.Data[url.Name] = strings.ReplaceAll(revised.Data[url.Name], url.Value, "")
+					if strings.HasPrefix(revised.Data[url.Name], ",") {
```

Comment:
The string operations seem to be repeated code. Could be moved into a different function.

Reply:
Same as above. TODO